src/Fmt_ast.ml

```diff
@@ -1903,7 +1903,11 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext ({ast= exp} as xexp)
                             | Pexp_letmodule _ | Pexp_object _ )
                         ; pexp_attributes= [] } as e1 )
                     , _ ) } as str ) ] )
-    when List.is_empty pexp_attributes ->
+    when List.is_empty pexp_attributes
+         && ( Poly.(c.conf.extension_sugar = `Always)
+            || Source_code_position.ascending ext.loc.loc_start
+                 e1.pexp_loc.loc_start
+               > 0 ) ->
```

Comment:
LGTM, though I would always like these things that are sensitive to the concrete syntax to be helpers in `Source` or similar.